src/rtctools_heat_network/workflows/io/write_output.py

```diff
@@ -689,6 +743,137 @@ def _name_to_asset(name):
+            variables_two_hydraulic_system = [
+                "Primary.HeatIn.Q",
+                "Primary.HeatIn.H",
+                "Secondary.HeatIn.Q",
+                "Secondary.HeatIn.H",
+                "Heat_flow",
+            ]
+            profiles = ProfileManager()
+            profiles.profile_type = "DATETIME_LIST"
+            profiles.profile_header = ["datetime"]
+
+            for ii in range(len(self.times())):
+                data_row = [self.io.datetimes[ii]]
+
+                for asset_name in [
+                    *self.heat_network_components.get("source", []),
+                    *self.heat_network_components.get("demand", []),
+                    *self.heat_network_components.get("pipe", []),
+                    *self.heat_network_components.get("buffer", []),
+                    *self.heat_network_components.get("ates", []),
+                    *self.heat_network_components.get("heat_exchanger", []),
+                    *self.heat_network_components.get("heat_pump", []),
+                ]:
+                    try:
+                        # If the asset has been placed
+                        asset = _name_to_asset(asset_name)
+
+                        try:
+                            # For all components dealing with one hydraulic system
+                            if isinstance(
+                                results[f"{asset_name}." + variables_one_hydraulic_system[0]][ii],
+                                numbers.Number,
+                            ):
+                                variables_names = variables_one_hydraulic_system
+                        except Exception:
+                            # For all components dealing with two hydraulic system
+                            if isinstance(
+                                results[f"{asset_name}." + variables_two_hydraulic_system[0]][ii],
+                                numbers.Number,
+                            ):
+                                variables_names = variables_two_hydraulic_system
+
+                        for variable in variables_names:
+                            if ii == 0:
+                                # Set header for each column
+                                profiles.profile_header.append(asset_name + "_" + variable)
+
+                                # Set profile database attributes for the esdl asset
+                                profile_attributes = esdl.InfluxDBProfile(
+                                    database=self.influxdb_database,
+                                    measurement=energy_system.id,
+                                    field=profiles.profile_header[-1],
+                                    port=self.influxdb_port,
+                                    host=self.influxdb_host,
+                                )
+                                asset.port[1].profile.append(profile_attributes)
```

Comment:
Is there a reason for the index 1 for the port?

Reply:
For now the outport is used to assign all results profile data to. Some code has been added to determine the outport index of the esdl asset

Reply:
The code has been updated to assign to the 1st outport found for the asset.